cmd/util/trace.go

```diff
@@ -0,0 +1,58 @@
+package util
+
+import (
+	"context"
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+	"go.opentelemetry.io/otel/trace"
+	"google.golang.org/grpc"
+)
+
+func InitTracer(serviceName, otlpAddress string) (trace.TracerProvider, func(), error) {
+	ctx := context.Background()
```

Comment:
We should probably receive the context from the main server build the resource with it.

Reply:
Updated.

Reply:
Sorry but it seems that this is still not using the context from the server. 
This is the context that I was referring to:
https://github.com/argoproj/argo-cd/blob/7beb8b6d5c841c8a98bc2e62f467755729ceb341/cmd/argocd-server/commands/argocd_server.go#L168

Reply:
Sorry I missed that. One question here, why do we need the for loop? https://github.com/argoproj/argo-cd/blob/7beb8b6d5c841c8a98bc2e62f467755729ceb341/cmd/argocd-server/commands/argocd_server.go#L166, can I move the ctx out of the loop because I don't want to put the init tracer function in the loop

Reply:
Thanks for the detailed reply. Then I think this is fixed.

Reply:
Sorry but you can't move the ctx out of the for-loop. A new context need to be build on every iteration and the `initTracer` call needs to happen somewhere inside this loop.